scripts/checks/verify_supply_chain.py

```
@@ -166,16 +334,230 @@ def verify_workflow_coverage() -> list[str]:
+        current_job_indent: int | None = None
+        workflow_defaults_indent: int | None = None
+        workflow_defaults_run_indent: int | None = None
+        job_defaults_indent: int | None = None
+        job_defaults_run_indent: int | None = None
+        in_jobs = False
+        step_working_directory: str | None = None
+        step_uses_workspace_exec = False
+
+        def record_step_violation() -> None:
+            effective_working_directory = (
+                step_working_directory
```

Ruff (0.15.12)

[warning] Function definition does not bind loop variable `step_working_directory`

(B023)